### PR TITLE
Ensure that the default initializers register last

### PR DESCRIPTION
**Pull request by weierophinney**
Per #58 and #64, 2.7.0 introduced a regression due to the fact that the default initializers, `injectFactory()` and `callElementInit()`, were registered *prior* to delegating to the parent constructor instead of *afterwards*. This led to problems when the order of initializers was important (e.g., if an initializer was practicing setter injection, and the injected instance was expected when `init()` was called by `callElementInit()`).

This patch provides a test that asserts the expected behavior, and updates the `FormElementManager` constructor to properly inject the initializers after calling the parent constructor.

### CODE REVIEW

**Comment by weierophinney** [comment]
After seeing the failing tests (all against v2 of zend-servicemanager), I did some digging.

Interestingly, in the v2 releases, `ServiceManager::addInitializer()` defines an optional, second parameter, `$topOfStack = true`. Internally, it uses that to determine if the initializer should be added to the start of the list (default) or end.

Prior to v2.7, `FormElementManager` did the following:

```php
$this->addInitializer([$this, 'injectFactory']);
$this->addInitializer([$this, 'callElementInit'], false);
```

In other words, it pushes the `injectFactory()` initializer to the top of the stack, but the `callElementInit()` factory to the end.

Updating the patch now to reflect this.

File: .travis.yml [hunk 1]
```diff
@@ -21,32 +23,42 @@ matrix:
...
     - php: 5.5
       env:
-        - EVENT_MANAGER_VERSION="^2.6.2"
-        - HYDRATOR_VERSION="^1.1"
-        - SERVICE_MANAGER_VERSION="^2.7.5"
+        - DEPS=locked
+        - CS_CHECK=true
+    - php: 5.5
+      env:
+        - DEPS=latest
```
Comment (weierophinney) [inline .travis.yml]: Switched to a lowest/locked/latest strategy, as it makes it far easier both on Travis and locally to switch between supported versions in order to test. (Which was critical for this patch, as I needed to verify behavior between zend-servicemanager major versions.)

File: composer.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
         "zendframework/zend-view": "^2.6.2",
         "zendframework/zendservice-recaptcha": "*",
         "fabpot/php-cs-fixer": "1.7.*",
-        "phpunit/PHPUnit": "~4.0"
+        "phpunit/phpunit": "^4.8"
```
Comment (weierophinney) [inline composer.json]: This was necessary as otherwise the "lowest" strategy failed due to usage of Prophecy in test cases.

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:noNamespaceSchemaLocation="./vendor/phpunit/phpunit/phpunit.xsd"
-         bootstrap="./test/bootstrap.php"
+         bootstrap="./vendor/autoload.php"
```
Comment (weierophinney) [inline phpunit.xml.dist]: The old test bootstrap was guarding against PHP version, which isn't necessary, and performing autoloading; it makes as much or better sense to just use the composer autoloader.

File: src/FormElementManager.php [hunk 4]
```diff
@@ -209,10 +209,11 @@ class FormElementManager extends AbstractPluginManager
      */
     public function __construct($configInstanceOrParentLocator = null, array $v3config = [])
     {
-        $this->initializers[] = [$this, 'injectFactory'];
-        $this->initializers[] = [$this, 'callElementInit'];
-
-        parent::__construct($configInstanceOrParentLocator, $v3config);
+        if (method_exists($this, 'configure')) {
```
Comment (weierophinney) [inline src/FormElementManager.php]: `configure` is only defined in v3, making this a reliable way to do version detection.
